Update list of mods using Polymer

unilock opened the pull request:
This should be all of them.

Adds:
- Advanced Golems
- Amazing Chest
- Eldritch Mobs
- FlashFreeze
- Grappling Hook
- Grimoire
- LifeSteal
- Map Render
- Nox
- Nox: Renoxed
- Packy
- Polysit
- Power Networks
- Random Serverside Blocks
- ServerMobs
- Set Bonuses
- Spice of Fabric
- Tatercart
- ~~Taterzens~~
- Universal Shops
- ~~YetAnotherAuthMod~~

Also adds the latest supported Minecraft version of each mod (which is likely doomed to fall out of date), updates some mod links, and fixes and standardizes grammar in a lot of places.

unilock committed:
Update MODS.md

unilock committed:
MODS.md: Add "Nox: Renoxed"

Patbox commented:
Would remove YetAnotherAuthMod as it's more or less designed for piracy-enabled servers and Taterzens, as it isn't really polymer mod (used purely for compat with quilt, but could work just fine without polymer)

unilock commented:
> Would remove YetAnotherAuthMod as it's more or less designed for piracy-enabled servers...

Oops, yeah that probably shouldn't be there then.

> ...and Taterzens, as it isn't really polymer mod (used purely for compat with quilt, but could work just fine without polymer)

I was having some trouble deciding which mods to include in MODS.md out of *all* mods that declare a dependency on Polymer, but I guess the deciding factor would be whether the mod *always* depends on Polymer *when running server-side only,* as opposed to just depending on Polymer in general (a loose requirement) or depending on Polymer always (a strict requirement).
I do believe all other mods require Polymer when running entirely server-side, though, so I'll leave them in.

unilock committed:
MODS.md: Remove Taterzens + YetAnotherAuthMod

Taterzens: only needs Polymer for Quilt compat

YetAnotherAuthMod: basically only for piracy-enabled servers
